sdk/cosmos/azure-cosmos-tests/src/test/java/com/azure/cosmos/rx/WebExceptionRetryPolicyE2ETests.java

```diff
@@ -0,0 +1,303 @@
+                    .build()
+            )
+            .build();
+
+        CosmosFaultInjectionHelper
+            .configureFaultInjectionRules(
+                cosmosAsyncContainer,
+                Arrays.asList(addressRefreshDelayRule, serverGoneRule)).block();
+        try {
+            cosmosAsyncContainer
+                .readItem(newItem.getId(), new PartitionKey(newItem.getId()), TestItem.class)
+                .block();
+            fail("addressRefreshHttpTimeout() should fail due to addressRefresh timeout");
+        } catch (CosmosException e) {
+            System.out.println("dataPlaneRequestHttpTimeout() Diagnostics " + " " + e.getDiagnostics());
+            assertThat(e.getStatusCode()).isEqualTo(HttpConstants.StatusCodes.REQUEST_TIMEOUT);
+            assertThat(e.getSubStatusCode()).isEqualTo(HttpConstants.SubStatusCodes.GATEWAY_ENDPOINT_READ_TIMEOUT);
+            validateAddressRefreshRetryPolicyResponseTimeouts(e.getDiagnostics());
+        } finally {
+            addressRefreshDelayRule.disable();
+            serverGoneRule.disable();
+        }
+    }
+
+    @Test(groups = {"multi-master"}, dataProvider = "operationTypeProvider", timeOut = 8 * TIMEOUT)
+    public void dataPlaneRequestHttpTimeout(
+        FaultInjectionOperationType faultInjectionOperationType,
+        OperationType operationType) {
+
+        if (BridgeInternal
+            .getContextClient(this.cosmosAsyncClient)
+            .getConnectionPolicy()
+            .getConnectionMode() != ConnectionMode.GATEWAY) {
+            throw new SkipException("queryPlanHttpTimeoutWillNotMarkRegionUnavailable() is only meant for GATEWAY mode");
+        }
+
+        TestItem newItem = TestItem.createNewItem();
+        this.cosmosAsyncContainer.createItem(newItem).block();
+        FaultInjectionRule requestHttpTimeoutRule = new FaultInjectionRuleBuilder("requestHttpTimeoutRule" + UUID.randomUUID())
+            .condition(
+                new FaultInjectionConditionBuilder()
+                    .operationType(faultInjectionOperationType)
+                    .connectionType(FaultInjectionConnectionType.GATEWAY)
+                    .build())
+            .result(
+                FaultInjectionResultBuilders.getResultBuilder(FaultInjectionServerErrorType.RESPONSE_DELAY)
+                    .delay(Duration.ofSeconds(66))
+                    .times(4) // make sure it will exhaust the local region retry
+                    .build()
+            )
+            .build();
+
+        CosmosFaultInjectionHelper.configureFaultInjectionRules(this.cosmosAsyncContainer, Arrays.asList(requestHttpTimeoutRule)).block();
+
+
+        try {
+            CosmosDiagnostics cosmosDiagnostics =
+                this.performDocumentOperation(cosmosAsyncContainer, operationType, newItem).block();
+            System.out.println("dataPlaneRequestHttpTimeout() Diagnostics " + " " + cosmosDiagnostics);
```

Comment:
Please remove the sys out log lines. If needed, use a logger.

Reply:
Done